src/transformers/models/maskformer/modeling_maskformer.py

```diff
@@ -0,0 +1,2463 @@
+# coding=utf-8
+# Copyright 2022 Facebook AI Research and The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PyTorch MaskFormer model."""
+
+from __future__ import annotations
+
+import collections.abc
+import logging
+import math
+import random
+from dataclasses import dataclass
+from numbers import Number
+from optparse import Option
+from typing import Dict, List, Optional, Tuple
+
+import numpy as np
+import torch
+import torch.distributed as dist
+from torch import Tensor, nn
+from torch.nn.functional import binary_cross_entropy_with_logits, cross_entropy, interpolate
+
+from ...activations import ACT2FN
+from ...file_utils import (
+    ModelOutput,
+    add_code_sample_docstrings,
+    add_start_docstrings,
+    add_start_docstrings_to_model_forward,
+    is_scipy_available,
+    requires_backends,
+)
+from ...modeling_outputs import BaseModelOutput, BaseModelOutputWithCrossAttentions
+from ...modeling_utils import PreTrainedModel, find_pruneable_heads_and_indices, prune_linear_layer
+from ..detr import DetrConfig
+from ..swin import SwinConfig
+from .configuration_maskformer import MaskFormerConfig
+
+
+if is_scipy_available():
+    from scipy.optimize import linear_sum_assignment
+
+logger = logging.get_logger(__name__)
+
+
+_CONFIG_FOR_DOC = "MaskFormerConfig"
+_CHECKPOINT_FOR_DOC = "facebook/Francesco/maskformer-swin-base-ade"
+_FEAT_EXTRACTOR_FOR_DOC = "MaskFormerFeatureExtractor"
+
+MASKFORMER_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "facebook/Francesco/maskformer-swin-base-ade",
+    # See all MaskFormer models at https://huggingface.co/models?filter=maskformer
+]
+
+
+# TODO ask what I should do with dist code
+def get_world_size() -> int:
+    if not dist.is_available():
+        return 1
+    if not dist.is_initialized():
+        return 1
+    return dist.get_world_size()
```

Comment:
We removed distributed code from DETR and LayoutLMv2 (because I had to after PR review), but honestly it's a pain afterwards for people to train it in a distributed setup.

So I'd argue to keep it. cc @sgugger